### PR TITLE
enforce random sampling at first call in kornia parallell transforms

### PR DESCRIPTION
**Pull request by edgarriba**
## What does this PR do?

enables to reuse the transforms parameters for kornia transforms in different calls

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/core/data/transforms.py [hunk 1]
```diff
@@ -73,6 +73,14 @@ class KorniaParallelTransforms(nn.Sequential):
 
     def __init__(self, *args):
         super().__init__(*[convert_to_modules(arg) for arg in args])
+        self._reuse_params: bool = False
+
+    @property
+    def reuse_params(self) -> bool:
+        return self._reuse_params
+
+    def set_reuse_params(self, val: bool):
```
Comment (ethanwharris) [inline flash/core/data/transforms.py:82]: Why not use `@reuse_params.setter` here?

**Review by ethanwharris — COMMENTED** [review]
Looks good, some failing tests. Is there any reason anyonme would want to do this though? It looks like `reuse_params=True` will make it apply the same transform every time, so you're not really getting an augmentation just a constant transform.

File: flash/core/data/transforms.py (hunk 1 above)
Comment (edgarriba) [reply flash/core/data/transforms.py:82]: if we don't have this e.g in segmentation I couldn't find a way to keep parameters to be applied to both images and masks. ok, for setter

Comment (edgarriba) [reply flash/core/data/transforms.py:82]: for the failing tests no idea - cannot reproduce in local

Comment (ethanwharris) [reply flash/core/data/transforms.py:82]: Looks like you need to update the segmentation model on AWS. But the usage here applies the same transform to image and mask no? https://github.com/PyTorchLightning/lightning-flash/blob/c11498d0c12357f448dedea0fa61262714f4fa11/flash/image/segmentation/transforms.py#L55

Comment (ethanwharris) [reply flash/core/data/transforms.py:82]: We definitely don't want to just always apply the same transform though

Comment (edgarriba) [reply flash/core/data/transforms.py:82]: ok, I was assuming for the case to call multiple times `KorniaParallelTransforms`

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

File: flash/image/segmentation/transforms.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def default_transforms(image_size: Tuple[int, int]) -> Dict[str, Callable]:
         "post_tensor_transform": nn.Sequential(
             ApplyToKeys(
                 [DefaultDataKeys.INPUT, DefaultDataKeys.TARGET],
-                KorniaParallelTransforms(K.geometry.Resize(image_size, interpolation='bilinear')),
+                KorniaParallelTransforms(K.geometry.Resize(image_size, interpolation='nearest')),
```
Comment (edgarriba) [inline flash/image/segmentation/transforms.py:41]: related to https://github.com/PyTorchLightning/lightning-flash/pull/352

**Pull request merged**